File: src/main/java/cubesrv/Responder.java
package cubesrv;

public abstract class Responder {
    public enum MessageType {
        MT_UNQUALIFIED,
        MT_PROGRESS,
        MT_MOVECOUNT,
        MT_SOLUTION
    }

    public abstract void handleMessage(MessageType mt, String msg);

    public void message(String msg) {
        handleMessage(MessageType.MT_UNQUALIFIED, msg);
    }

    public void progress(String msg) {
        handleMessage(MessageType.MT_PROGRESS, msg);
    }

    public void movecount(String msg) {
        handleMessage(MessageType.MT_MOVECOUNT, msg);
    }

    public void solution(String msg) {
        handleMessage(MessageType.MT_SOLUTION, msg);
    }
}

File: src/main/kotlin/cubesrv/Responder.kt
package cubesrv

abstract class Responder {
    enum class MessageType {
        MT_UNQUALIFIED,
        MT_PROGRESS,
        MT_MOVECOUNT,
        MT_SOLUTION
    }
    abstract fun handleMessage(mt : MessageType, msg : String)

    fun message(msg : String) : Unit {
        handleMessage(MessageType.MT_UNQUALIFIED, msg);
    }

    fun progress(msg : String) : Unit {
        handleMessage(MessageType.MT_PROGRESS, msg);
    }

    fun movecount(msg : String) : Unit {
        handleMessage(MessageType.MT_MOVECOUNT, msg);
    }

    fun solution(msg : String) : Unit {
        handleMessage(MessageType.MT_SOLUTION, msg);
    }
}

